rct229/ruleset_functions/get_zone_conditioning_category_dict.py

```diff
@@ -0,0 +1,228 @@
+from rct229.data_fns.table_3_2_fns import table_3_2_lookup
+from rct229.ruleset_functions.get_opaque_surface_type import get_opaque_surface_type
+from rct229.schema.config import ureg
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import pint_sum
+
+CAPACITY_THRESHOLD = 3.4 * ureg("Btu/(hr * ft2)")
+CRAWLSPACE_HEIGHT_THRESHOLD = 7 * ureg("ft")
+ZERO_AREA = 0 * ureg("ft2")
+ZERO_UA = 0 * ureg("ft2 * Btu / (hr * ft2 * R)")
+
+
+def get_zone_conditioning_category_dict(climate_zone, building):
+    """Determines the zone conditioning category for every zone in a building
+
+    Parameters
+    ----------
+    climate_zone : str
+        One of the ClimateZone2019ASHRAE901 enumerated values
+    building : dict
+        A dictionary representing a building as defined by the ASHRAE229 schema
+    Returns
+    -------
+    dict
+        A dictionary that maps zones to one of the conditioning categories:
+        "CONDITIONED MIXED", "CONDITIONED NON-RESIDENTIAL", "CONDITIONED RESIDENTIAL",
+        "SEMI-HEATED", "UNCONDITIONED", "UNENCLOSED"
+    """
+    system_min_heating_output = table_3_2_lookup(climate_zone)[
+        "system_min_heating_output"
+    ]
+
+    zone_conditioning_category_dict = {}
+
+    building_zones = find_all("building_segments[*].zones[*]", building)
+
+    building_hvac_systems = find_all(
+        "building_segments[*].heating_ventilation_air_conditioning_systems[*]",
+        building,
+    )
+    # Create a map from an hvac system's id to the hvac system itself
+    building_hvac_systems_map = dict(
+        [(hvac_system["id"], hvac_system) for hvac_system in building_hvac_systems]
+    )
+
+    # Determine eligibility for directly conditioned (heated or cooled) and
+    # semi-heated zones
+    directly_conditioned_zone_ids = []
+    semiheated_zone_ids = []
+    for zone in building_zones:
+        zone_id = zone["id"]
+
+        zone_hvac_systems = [
+            building_hvac_systems_map[hvac_system_id]
+            for hvac_system_id in zone[
+                "served_by_heating_ventilation_air_conditioning_systems"
```

Comment:
Possible change in the schema from zone.served_by_heating_ventilation_air_conditioning_systems to zone.terminal.served_by_heating_ventilation_air_conditioning_systems but has not been finalized yet.